--- build-old/make/CompileDemos.gmk	2022-05-18 16:24:10.034165701 +0100
+++ build/make/CompileDemos.gmk	2022-05-18 16:23:34.639162860 +0100
@@ -232,7 +232,7 @@
 
 $(SUPPORT_OUTPUTDIR)/demos/image/nbproject/%: $(DEMO_SHARE_SRC)/nbproject/%
 	$(call install-file)
-	$(CHMOD) -f ug+w $@
+	$(CHMOD) ug+w $@
 
 ifeq ($(OPENJDK_TARGET_OS), solaris)
   TARGETS += $(patsubst $(DEMO_SHARE_SRC)/nbproject/%, \
--- build-old/make/common/JavaCompilation.gmk	2022-05-18 16:24:10.034165701 +0100
+++ build/make/common/JavaCompilation.gmk	2022-05-18 16:23:34.639162860 +0100
@@ -78,7 +78,7 @@
     $$($1_BIN)$$($1_MODULE_SUBDIR)$$($2_TARGET) : $2
 	$(MKDIR) -p $$(@D)
 	$(CP) $$< $$@
-	$(CHMOD) -f ug+w $$@
+	$(CHMOD) ug+w $$@
 
     # And do not forget this target
     $1_ALL_COPY_TARGETS += $$($1_BIN)$$($1_MODULE_SUBDIR)$$($2_TARGET)
@@ -130,7 +130,7 @@
 	        -e 's/^[ 	]*//;s/[ 	]*$$$$//' \
 	        -e 's/\\=/=/' \
 	    | $(SORT) > $$@
-	$(CHMOD) -f ug+w $$@
+	$(CHMOD) ug+w $$@
 
     # And do not forget this target
     $1_ALL_COPY_CLEAN_TARGETS += $$($1_BIN)$$($1_MODULE_SUBDIR)$$($2_TARGET)
