--- build/src/3rdparty/chromium/mojo/public/cpp/platform/named_platform_channel_posix.cc	2022-07-07 22:23:50.602954227 +0100
+++ build2/src/3rdparty/chromium/mojo/public/cpp/platform/named_platform_channel_posix.cc	2022-07-11 12:57:26.005060552 +0100
@@ -8,6 +8,7 @@
 #include <sys/socket.h>
 #include <sys/un.h>
 #include <unistd.h>
+#include <cstring>
 
 #include "base/files/file_util.h"
 #include "base/files/scoped_file.h"
